src/core/utils/CorePlugin.ts

```diff
@@ -34,17 +35,29 @@ export class CorePlugin extends JWPlugin {
     // Public
     //--------------------------------------------------------------------------
 
-    enter(): void {
-        this.editor.vDocument.enter();
+    /**
+     * Handle the (shift+) enter key.
+     *
+     * @param intent
+     */
+    enter(intent: KeydownIntent): void {
+        if (intent.payload.shiftKey) {
+            this.editor.vDocument.insert(new VNode(VNodeType.LINE_BREAK));
+        } else {
+            this.editor.vDocument.enter();
+        }
```

Comment:
No. Enter and Shift+Enter trigger are two different intents and so two different commands. I think Shift+Enter is `insertLineBreak`, so no need for a "keyboard intent" which makes no sense in our abstraction IMO.

Reply:
Ok, makes sense. Should be handled in EventManager then?

Reply:
Yes, see my above comment about the keymap.